block-producer/src/batch_builder/batch.rs

```diff
@@ -113,6 +127,17 @@ impl TransactionBatch {
     pub fn created_notes_root(&self) -> Digest {
         self.created_notes_smt.root()
     }
+
+    fn calc_id(txs: &[SharedProvenTx]) -> BatchId {
+        let mut result = BatchId::default();
+
+        for tx in txs {
+            let tx_hash = Blake3_256::hash_elements(tx.id().inner().as_elements());
+            result = Blake3_256::merge(&[result, tx_hash]);
+        }
```

Comment:
The way this look currently works will result in quite a bit of unnecessary hashing. What I would probably is:

1. Build a byte vector (using `tx.id().as_bytes`).
2. Hash this vector using `Blake3_256::hash()`.

My guess is that this will reduce the amount of hashing by about 3x.